src/util/firstNotNull.js

```diff
@@ -1,16 +1,21 @@
+import { isFunction } from '.';
+
 /**
  * Return the first argument whose value is not null
  * @param args
  * @returns {*}
  */
-let firstNotNull = function firstNotNull(...args) {
-  while(args && args.length > 0) {
+function firstNotNull(...args) {
+  while (args && args.length > 0) {
     let next = args.shift();
-    if( next != null){
+    if (isFunction(next)) {
+      next = next();
```

Comment:
here it's going to trigger `next()`, isn't the function supposed to return the first param not `null` ?